### PR TITLE
Bugfix/814 handle multi factor data graphs

### PR DESCRIPTION
**Pull request by ppratikcr7**
In this PR, I am handling main effect and interaction effect graphs or table.

Now for more than 2 factors, we will have a horizontal scrolling main effect graphs. For interaction effect, we have a line graph for 2 factors and a table for more than 2 factors.

### CODE REVIEW

**Comment by danoswaltCL** [comment]
This needs to go for the release branch also? Relates to https://github.com/CarnegieLearningWeb/UpGrade/issues/814 @amurphy-cl @mswartzCL ?

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/experiment-query-result/experiment-query-result.component.ts [hunk 1]
```diff
@@ -29,20 +44,27 @@ export class ExperimentQueryResultComponent implements OnInit, OnDestroy {
...
   experimentType: string = null;
   data: { name: string; series: { name: string; value: number }[]; dot: boolean }[];
   meanData2: { name: string; value: number }[];
   meanData1: { name: string; value: number }[];
   maxLevelCount = 0;
+  factorColumnDefs: FactorColumnDef[] = [];
+  queryColumnDefs: QueryColumnDef[] = [];
+  dataSource: RowData[] = [];
+
+  // dataSource = [
```
Comment (danoswaltCL) [inline frontend/projects/upgrade/src/app/features/dashboard/home/components/experiment-query-result/experiment-query-result.component.ts:64]: this looks like some commented old debug code, can it be removed?

Comment (ppratikcr7) [reply frontend/projects/upgrade/src/app/features/dashboard/home/components/experiment-query-result/experiment-query-result.component.ts]: Oh ...missed that..thanks..Also updated few things for graphs UI polishing in the next commit.

**Comment by danoswaltCL** [comment]
Does this need to go for the release branch also? Relates to https://github.com/CarnegieLearningWeb/UpGrade/issues/814 @amurphy-cl @mswartzCL ?

**Comment by ppratikcr7** [comment]
> Does this need to go for the release branch also? Relates to #814 @amurphy-cl @mswartzCL ?

@danoswaltCL Just wanted to pitch in. We already have the [commits](https://github.com/CarnegieLearningWeb/UpGrade/commit/ee41558a7b8301320c8e8d8705e9908798bd3294) for multi factor support in our release branch. If this PR is not merged in release branch, then we would not have proper graphs and layout for 3 or more factors, in case anyone is attempting to do so.

**Comment by danoswaltCL** [comment]
per the mtg discussion, plz reopen changes against `dev` instead of `release/v4.1`

**Comment by ppratikcr7** [comment]
> per the mtg discussion, plz reopen changes against `dev` instead of `release/v4.1`

Done

**Review by jreddig — APPROVED** [review]

**Pull request merged**